components/Account/Access/Campaigns/Form.js

```diff
@@ -0,0 +1,168 @@
+import React, { Component, Fragment } from 'react'
+import isEmail from 'validator/lib/isEmail'
+
+import { Button, Field, Label, InlineSpinner, Interaction, A } from '@project-r/styleguide'
+
+import { timeFormat } from '../../../../lib/utils/format'
+
+import ErrorMessage from '../../../ErrorMessage'
+
+const { H3, P } = Interaction
+
+const dayFormat = timeFormat('%e. %B %Y')
+
+class Form extends Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      email: '',
+      error: false,
+      isMutating: false,
+      hasMutated: false,
+      mutationError: null,
+      dirty: {},
+      grantedLast: {}
+    }
+
+    this.hasMutated = ({ data }) => {
+      this.setState({
+        isMutating: false,
+        hasMutated: true,
+        grantedLast: {
+          email: data.grantAccess.email,
+          endAt: data.grantAccess.endAt
+        }
+      })
+    }
+
+    this.catchMutationError = error => {
+      this.setState({
+        isMutating: false,
+        mutationError: error
+      })
+    }
+
+    this.onClickReset = (event, value, shouldValidate) => {
+      event.preventDefault()
+
+      this.setState({
+        value: '',
+        hasMutated: false,
+        dirty: {}
+      })
+    }
+
+    this.onChangeEmail = (event, value, shouldValidate) => {
+      event.preventDefault()
+
+      this.setState(() => ({
+        value,
+        error: (
+          value.trim().length <= 0 &&
+          'Keine E-Mail-Adresse angegeben'
+        ) || (
+          !isEmail(value) &&
+          'Keine gültige E-Mail-Adresse erkannt'
+        ),
+        dirty: {
+          email: shouldValidate
+        }
+      }))
+    }
+
+    this.onSubmit = async (event) => {
+      event.preventDefault()
+
+      this.setState({
+        isMutating: true,
+        mutationError: false
+      })
+
+      return this.props.grantAccess({
+        campaignId: this.props.campaign.id,
+        email: this.state.value
+      })
+        .then(this.hasMutated)
+        .catch(this.catchMutationError)
+    }
+  }
+
+  render () {
+    const { campaign } = this.props
+
+    if (campaign.slots.free === 0) {
+      if (campaign.slots.total > 1) {
+        return (
+          <Label>Alle verfügbaren Zugriffe vergeben.</Label>
+        )
+      }
+
+      return null
+    }
+
+    const {
+      value,
+      error = false,
```

Comment:
This is redundant and unnecessary. Redundant because already false in initial state via construction. Unnecessary because error undefined would work fine.